sql/sql_yacc.yy

```diff
@@ -7882,7 +7890,7 @@ opt_ev_sql_stmt:
         ;
 
 ident_or_empty:
-          /* empty */ { $$= Lex_ident_sys(); }
+          /* empty */ %prec PREC_BELOW_IDENTIFIER_OPT_SPECIAL_CASE { $$= Lex_ident_sys(); }
```

Comment:
I don't really like this and I believe it doesn't cover the case when database is named `comment` and we try to alter it without specifying it's name:

ex:
```
use comment;
alter database comment='foo'
```

I'd prefer if we didn't have to add special precedence rules, if possible. I'm currently experimenting with removing ident_or_empty entirely and have 2 separate branches for `ALTER DATABASE ident create_database_options` and `ALTER DATABASE create_database_options`

If you find a solution that works with this, feel free to propose it.

Reply:
The failing scenario applies to other database names as well, as discussed in the previous comment.

I had to set the precedence of `COMMENT_SYM` higher than `empty`, to avoid a shift/reduce conflict. But if there is a better way to avoid this conflict, please let me know.

The `ALTER DATABASE` without the `ident` works fine for `CHARACTER SET` and `COLLATE`, but not for `COMMENT`. I am not sure right now how this is happening. I will also be looking into this.